cmd/root/root.go

```diff
@@ -109,6 +114,38 @@ func rootPersistentPreRunEFunc(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	}
+
+	// Handle chunk size flags (deprecation and overwriting where needed)
+	//
+	// Expected Behavior for chunk size flags:
+	// * If either cli option is set, use that value
+	// * If both are set, emit an error
+	// * If neither are set, use ENV values
+	// ** If PGET_CHUNK_SIZE is set, use that value
+	// ** If PGET_CHUNK_SIZE is not set, use PGET_MINIMUM_CHUNK_SIZE if set
+	//    NOTE: PGET_MINIMUM_CHUNK_SIZE value is just set over the key for PGET_CHUNK_SIZE
+	//    Warning message will be emitted
+	// ** If both PGET_CHUNK_SIZE and PGET_MINIMUM_CHUNK_SIZE are set, use PGET_CHUNK_SIZE
+	//    Warning message will be emitted
+	// * If neither are set, use the default value
+
+	changedMin := cmd.PersistentFlags().Changed(config.OptMinimumChunkSize)
+	changedChunk := cmd.PersistentFlags().Changed(config.OptChunkSize)
+	if changedMin && changedChunk {
+		return fmt.Errorf("--minimum-chunk-size and --chunk-size cannot be used at the same time, use --chunk-size instead")
+	} else if !(changedMin && changedChunk) {
+		minChunkSizeEnv := viper.GetString(config.OptMinimumChunkSize)
+		chunkSizeEnv := viper.GetString(config.OptChunkSize)
+		if minChunkSizeEnv != chunkSizeDefault {
+			if chunkSizeEnv == chunkSizeDefault {
+				logger.Warn().Msg("Using PGET_MINIMUM_CHUNK_SIZE is deprecated, use PGET_CHUNK_SIZE instead")
+				viper.Set(config.OptChunkSize, minChunkSizeEnv)
+			} else {
+				logger.Warn().Msg("Both PGET_MINIMUM_CHUNK_SIZE and PGET_CHUNK_SIZE are set, using PGET_CHUNK_SIZE")
+			}
+		}
+	}
```

Comment:
Do we need to do something similar with PGET_MAX_CHUNKS and PGET_CONCURRENCY? We've removed the alias but we haven't done anything else to maintain the alias beyond binding both to the same variable (which you've established doesn't handle the environment variable case).

Reply:
I'm going to assume "no" because we don't set these envvars in cluster and models are probably using the command line args